### PR TITLE
Add Navigator documentation

### PR DESCRIPTION
**Pull request by MatthewHawkins**

### CODE REVIEW

File: docs/components/navigator.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+#### Page size
+
+The `setSize()` method allows you to specify the number of items to display on each page of the pagination. When you call this method and provide a new page size, it adjusts the pagination accordingly. 
+
+```java
+paginator.setSize(pageSize);
+```
+
+<ComponentDemo 
+path='https://demo.webforj.com/webapp/controlsamples/NavigatorPages?'
```
Comment (hyyan) [inline docs/components/navigator.md:103]: isn't this the same demo the previous one ?

File: docs/components/navigator.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+For example, to set the text of the last page button in a `Navigator` with 10 pages to "Go to page 10", use the following code snippet: 
+<!-- vale on -->
+
+```java
+navigator.setText("'Go to page ' + endPage", Navigator.Part.LAST_BUTTON);
+```
+
+### Tooltip text
+
+You can customize tooltips for various parts of the Navigator component using the setTooltipText() method. Tooltips provide helpful hints to users when they hover over navigation elements.
```
Comment (hyyan) [inline docs/components/navigator.md:138]: setTooltipText() should be in backticks

File: docs/components/navigator.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+It supports automatic disabling of navigation buttons based on the current page and total items, and offers customization options for text and tooltips for different parts of the navigator. Additionally, you can bind it to a `Paginator` instance to manage the data set's pagination logic and reflect changes in the navigation controls. 
+
+## Binding to repositories
+
+Often, a `Navigator` component displays information found in a bound `Repository`. This binding enables the `Navigator` to automatically paginate data managed by the repository and refresh other bindable components, such as tables, based on the navigated data.
+
+To do this, simply pass the desired `Repository` object to an applicable `Navigator` object's constructor:
+
+<ComponentDemo 
+path='https://demo.webforj.com/webapp/controlsamples/NavigatorTable?'
```
Comment (hyyan) [inline docs/components/navigator.md:38]: This demo is not rendering correctly

File: docs/components/navigator.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+navigator.setLayout(Navigator.Layout.QUICK_JUMP);
+```
+
+<ComponentDemo 
+path='https://demo.webforj.com/webapp/controlsamples/NavigatorLayout?' 
+javaE='https://raw.githubusercontent.com/webforj/ControlSamples/main/src/main/java/componentdemos/navigatordemos/NavigatorLayout.java'
+height='200px'
+/>
+
+## Events
```
Comment (hyyan) [inline docs/components/navigator.md:210]: Remove [https://github.com/webforj/webforj-docs/issues/121](https://github.com/webforj/webforj-docs/issues/121)

File: docs/components/navigator.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+Similarly, you can retrieve the index of the last item on the current page with `getEndIndex()`.
+
+```java
+int endIndex = paginator.getEndIndex();
+```
+Now, developers can use these indices to fetch and display the relevant items within the current page.
+
+### Pages
+
+The concept of "pages" in the `Paginator` refers to the logical divisions of the entire dataset. Each page contains a specific number of items, and users navigate through these pages to explore the complete dataset.
```
Comment (hyyan) [inline docs/components/navigator.md:82]: remove double quotes around pages

File: docs/components/navigator.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+```java
+navigator.setTooltipText("Go to the last page", Navigator.Part.LAST_BUTTON);
+```
+
+In the following example, the `setText()` method displays a numeric value to the user. Clicking the buttons fires the `onChange` method of the `Navigator`, which comes with a `Direction` value the clicked button. 
+
+See [this section](#events) for more details on the available `Navigator` events. 
+
+<ComponentDemo 
+path='https://demo.webforj.com/webapp/controlsamples/NavigatorBasic?'
```
Comment (hyyan) [inline docs/components/navigator.md:157]: Wrong demo

File: docs/components/navigator.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+
+#### 4. Quick jump layout
+
+The quick-jump layout provides a [NumberField](./fields/number-field.md) for users to enter a page number for quick navigation. This is useful when users need to navigate to a specific page quickly, especially for large datasets To activate this layout, use:
+
+```java
+navigator.setLayout(Navigator.Layout.QUICK_JUMP);
+```
+
+<ComponentDemo
```
Comment (hyyan) [inline docs/components/navigator.md:204]: This demo looks like it is for  Quick jump layout, move it to the top

**Review by hyyan — CHANGES_REQUESTED** [review]

File: docs/components/navigator.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+import JavadocLink from '@site/src/components/DocsTools/JavadocLink';
+import DocChip from '@site/src/components/DocsTools/DocChip';
+import FiberSmartRecordIcon from '@mui/icons-material/FiberSmartRecord';
+import Chip from '@mui/material/Chip';
+
+<!-- vale on -->
+
+<DocChip tooltipText="This component will render with a shadow DOM, an API built into the browser that facilitates encapsulation." label="Shadow" component="a" href="../glossary#shadow-dom" target="_blank" clickable={true} iconName="shadow" />
+
+<DocChip tooltipText="The name of the web component that will render in the DOM." label="dwc-navigator" href="https://basishub.github.io/basis-next/#/web-components/dwc-button" clickable={false} iconName='code'/>
```
Comment (hyyan) [inline docs/components/navigator.md:21]: why it says dwc-button

File: docs/components/navigator.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+
+
+<JavadocLink type="engine" location="com/webforj/component/layout/navigator/Navigator" top='true'/>
```
Comment (hyyan) [inline docs/components/navigator.md:24]: this should be : com/webforj/component/navigator/Navigator

**Review by hyyan — CHANGES_REQUESTED** [review]

File: docs/components/navigator.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+For example, to set the tooltip text of the last page button in a `Navigator` to "Go to the last page", use the following code snippet:
+<!-- vale on -->
+
+```java
+navigator.setTooltipText("Go to the last page", Navigator.Part.LAST_BUTTON);
+```
+
+In the following example, the `setText()` method displays a numeric value to the user. Clicking the buttons fires the `onChange` method of the `Navigator`, which comes with a `Direction` value the clicked button. 
+
+See [this section](#events) for more details on the available `Navigator` events.
```
Comment (hyyan) [inline docs/components/navigator.md:161]: remove this too

File: docs/components/navigator.md (hunk 9 above)
Comment (hyyan) [reply docs/components/navigator.md]: The url still wrong , component is doubled in the URL

File: docs/components/navigator.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+
+<ComponentDemoMultiple 
+path='https://demo.webforj.com/webapp/controlsamples/NavigatorTable?' 
+javaE='https://raw.githubusercontent.com/webforj/ControlSamples/main/src/main/java/componentdemos/navigatordemos/NavigatorTable.java'
+height='500px'
+/>
+
+This example creates the `Navigator` and [`Table`](../addons/table/) with the same `Repository` instance. This means that when navigating to a new page with the `Navigator`, the [`Table`](../addons/table/) recognizes this change and rerenders.
+
+## Pagination
```
Comment (hyyan) [inline docs/components/navigator.md:45]: This section should explain that the navigator can accept a Paginator instance. When integrating with a Paginator, the navigator will respond to any changes within the Paginator. Additionally, this guide includes practical code snippets to illustrate how this integration works in practice.

File: docs/components/navigator.md [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+
+The `Navigator` component provides extensive customization options for buttons, text, and tooltips. To change the displayed text on the `Navigator` component, use the `setText()` method. This method takes text, as well as the desired `Part` of the `Navigator`.
+
+<ComponentDemoMultiple 
+path='https://demo.webforj.com/webapp/controlsamples/NavigatorBasic?' 
+javaE='https://raw.githubusercontent.com/webforj/ControlSamples/main/src/main/java/componentdemos/navigatordemos/NavigatorBasic.java'
+height='100px'
+/>
+
+### Button and component text
```
Comment (hyyan) [inline docs/components/navigator.md:99]: That should be : "Buttons ..." (plural)

File: docs/components/navigator.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+<TableBuilder tag={require('@site/docs/components/_dwc_control_map.json').Navigator} table="reflects" />
+<!-- vale on -->
+
+### Dependencies
+
+This component relies on the following components - see the related article for more detailed styling information:
+
+<TableBuilder tag='dwc-navigator' table="dependencies"/>
+
+## Best practices
```
Comment (hyyan) [inline docs/components/navigator.md:221]: please change to :

Understand dataset
Choose appropriate layout
Customize text and tooltips

**Review by hyyan — CHANGES_REQUESTED** [review]

File: docs/styling/getting_started.md [hunk 14]
```diff
@@ -2,7 +2,7 @@
 sidebar_position: 1
 displayed_sidebar: documentationSidebar
 id: "styling_intro"
-pagination_prev: "components/text-area"
+pagination_prev: "components/tabbed-pane"
```
Comment (hyyan) [inline docs/styling/getting_started.md:5]: why this change ?

File: src/components/DocsTools/ComponentDemoMultiple.js [hunk 15]
```diff
@@ -261,6 +263,8 @@ export default function ComponentDemo({
     width: 100%;
     height: ${height || "100%"};
     pointer-events: ${isResizing ? "none" : "auto"};
+    background-color: #F8FAFC;
+    padding: ${padding || "1em"};
```
Comment (hyyan) [inline src/components/DocsTools/ComponentDemoMultiple.js:267]: Move this outside the PR and send it in its own PR, this change is causing issues in splitter demos , test it more

File: docs/styling/getting_started.md (hunk 14 above)
Comment (MatthewHawkins) [reply docs/styling/getting_started.md:5]: Happed due to a rebase on website

File: src/components/DocsTools/ComponentDemoMultiple.js (hunk 15 above)
Comment (MatthewHawkins) [reply src/components/DocsTools/ComponentDemoMultiple.js]: Fixed this

**Review by hyyan — APPROVED** [review]

**Pull request merged**